datafusion/common/src/column.rs

```diff
@@ -70,7 +71,7 @@ impl Column {
         // name
         Column {
             relation: None,
-            name: String::from(flat_name),
+            name: flat_name,
```

Comment:
This would always clone previously, and now it allows the caller to pass in a `String` if they have one and avoid the `clone`